src/vm/analysis/arithmetic_checker/mod.rs

```diff
@@ -189,7 +189,11 @@ impl<'a> ArithmeticOnlyChecker<'a> {
             | Modulo | Power | Sqrti | Log2 | BitwiseXOR | And | Or | Not | Equals | If
             | ConsSome | ConsOkay | ConsError | DefaultTo | UnwrapRet | UnwrapErrRet | IsOkay
             | IsNone | Asserts | Unwrap | UnwrapErr | IsErr | IsSome | TryRet | ToUInt | ToInt
-            | Len | Begin | TupleMerge => self.check_all(args),
+            | BuffToIntLe | BuffToUIntLe | BuffToIntBe | BuffToUIntBe | Len | Begin
+            | TupleMerge => {
+                // Check all arguments.
+                self.check_all(args)
+            }
```

Comment:
I think that these buff conversion functions should not pass the arithmetic only checker -- so these new functions would be "FunctionNotPermitted".

Reply:
done